functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,302 @@
+                        $JobStep.Subsystem = $Subsystem
+                    }
+
+                    if ($Command) {
+                        Write-Message -Message "Setting job step command to $Command" -Level Verbose 
+                        $JobStep.Command = $Command
+                    }
+
+                    if ($CmdExecSuccessCode) {
+                        Write-Message -Message "Setting job step command exec success code to $CmdExecSuccessCode" -Level Verbose 
+                        $JobStep.CommandExecutionSuccessCode = $CmdExecSuccessCode
+                    }
+
+                    if ($OnSuccessAction) {
+                        Write-Message -Message "Setting job step success action to $OnSuccessAction" -Level Verbose 
+                        $JobStep.OnSuccessAction = $OnSuccessAction
+                    }
+
+                    if ($OnSuccessStepId) {
+                        Write-Message -Message "Setting job step success step id to $OnSuccessStepId" -Level Verbose 
+                        $JobStep.OnSuccessStep = $OnSuccessStepId
+                    }
+
+                    if ($OnFailAction) {
+                        Write-Message -Message "Setting job step fail action to $OnFailAction" -Level Verbose 
+                        $JobStep.OnFailAction = $OnFailAction
+                    }
+
+                    if ($OnFailStepId) {
+                        Write-Message -Message "Setting job step fail step id to $OnFailStepId" -Level Verbose 
+                        $JobStep.OnFailStep = $OnFailStepId
+                    }
+
+                    if ($DatabaseName) {
+                        # Check if the database is present on the server
+                        if (($Server.Databases).Name -contains $DatabaseName) {
+                            Write-Message -Message "Setting job step database name to $DatabaseName" -Level Verbose 
+                            $JobStep.DatabaseName = $DatabaseName
+                        }
+                        else {
+                            Stop-Function -Message ("The database is not present on instance $instance.") -Target $SqlServer 
+                            return
+                        }
+                    }
+
+                    if (($DatabaseUserName) -and ($DatabaseName)) {
+                        # Check if the username is present in the database
+                        if (($Server.Databases[$DatabaseName].Users).Name -contains $DatabaseUserName) {
+                            Write-Message -Message "Setting job step database username to $DatabaseUserName" -Level Verbose 
+                            $JobStep.DatabaseUserName = $DatabaseUserName
+                        }
+                        else {
+                            Stop-Function -Message ("The database user is not present in the database $DatabaseName on instance $instance.") -Target $SqlServer 
+                            return
```

Comment:
Instead of the interrupt version, here you want to use the continue implementation of stop function, as only processing of this item has failed. Also no need to wrap the string in braces here:
` Stop-Function -Message "The database user is not present in the database $DatabaseName on instance $instance." -Target $instance -Continue`